tests/Functional/BlobTest.php

```diff
@@ -169,6 +172,28 @@ public function testBindParamProcessesStream(): void
         $this->assertBlobContains('test');
     }
 
+    public function testBindValueResetsStream(): void
+    {
+        if (TestUtil::isDriverOneOf('oci8')) {
+            self::markTestIncomplete('The oci8 driver does not support stream resources as parameters');
+        }
+
+        $stmt = $this->connection->prepare(
+            "INSERT INTO blob_table(id, clobcolumn, blobcolumn) VALUES (1, 'ignored', ?)",
+        );
+
+        $stream = fopen('php://temp', 'rb+');
+        fwrite($stream, 'a test');
+        fseek($stream, 2);
+        $stmt->bindValue(1, $stream, ParameterType::LARGE_OBJECT);
+
+        $stmt->execute();
+
+        self::assertEquals(2, ftell($stream), 'Resource parameter should be reset to position before execute.');
+
+        $this->assertBlobContains('test');
```

Comment:
Curious. Do we expect a seekable stream to be copied from the position of the pointer or do we expect it to be copied from the beginning? This assertion would allow both results, `a test` as well as `test`.

Reply:
The expectation I'm working on is that the stream is copied from the current position to the end.

I'd be happy to use assertBlobEquals, but I don't think I saw it when I wrote the test.

Reply:
> I'd be happy to use assertBlobEquals, but I don't think I saw it when I wrote the test.

My bad, looking at the implementation of `assertBlobContains()`, it actually does what we're looking for.

https://github.com/doctrine/dbal/blob/96d5a70fd91efdcec81fc46316efc5bf3da17ddf/tests/Functional/BlobTest.php#L207